Plexdrive segfaults when trying to clean local chunks

Getting this:

```
[PLEXDRIVE] [2017-05-24 20:55] DEBUG  : Cleaning empty directory /home/<user>/plexdrive/tmp/chunks/0B4VMpKMUqD_SLU1DbXFGMDcwOWc
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x7c0b08]

goroutine 20161245 [running]:
main.deleteOldestFile.func1(0xc42078db80, 0x3e, 0x0, 0x0, 0xd646e0, 0xc42b8e6450, 0x0, 0x0)
        /go/src/github.com/dweidenfeld/plexdrive/buffer.go:271 +0x38
path/filepath.walk(0xc420187770, 0x21, 0xd6b2a0, 0xc4208cb380, 0xc421155940, 0x0, 0x20)
        /usr/local/go/src/path/filepath/path.go:372 +0x2fe
path/filepath.Walk(0xc420187770, 0x21, 0xc421155940, 0x8ed800, 0x0)
        /usr/local/go/src/path/filepath/path.go:398 +0x14c
main.deleteOldestFile(0xc420187770, 0x21, 0x9ffaad36f, 0x0)
        /go/src/github.com/dweidenfeld/plexdrive/buffer.go:279 +0x119
main.cleanChunkDir(0xc420187770, 0x21, 0x20, 0xc420572720)
        /go/src/github.com/dweidenfeld/plexdrive/buffer.go:257 +0x98
main.(*Buffer).ReadBytes.func2()
        /go/src/github.com/dweidenfeld/plexdrive/buffer.go:170 +0x3d
created by main.(*Buffer).ReadBytes
        /go/src/github.com/dweidenfeld/plexdrive/buffer.go:174 +0x1393
```

When the chunks directory passes over the specified `--clear-chunk-max-size` and gets removed, that error is thrown and plexdrive stops. The fuse mount doesn't get unmounted however and is broken.

--- a/buffer.go
+++ b/buffer.go
@@ -268,6 +268,14 @@ func deleteOldestFile(path string) error {
 	lastMod := time.Now()
 
 	err := filepath.Walk(path, func(file string, info os.FileInfo, err error) error {
+		if err != nil {
+			Log.Errorf("Error during walk through cache directory: %+v", err)
+			return filepath.SkipDir
+		}
+		if info == nil {
+			Log.Errorf("File info for %s was nil", file)
+			return filepath.SkipDir
+		}
 		if !info.IsDir() {
 			modTime := info.ModTime()
 			if modTime.Before(lastMod) {

